На данной практической работе я буду разбирать датасет о "Греческие Боги". Он хранит в себе информацию об именах на разных языках (Греческий и Английский), их типах, подтипах и их описании. С помощью определённых функций узнаю ифнормацию о данном датасете и проденстрирую это на деле. 

In [1]:
import pandas as pd
import numpy as np
greek_gods = pd.read_csv(r"C:\Users\user\Downloads\greek_gods.csv")
greek_gods.head()

Unnamed: 0 name-english name-greek main-type  sub-type  \
0           0    Aphrodite   Ἀφροδίτη  боженька  olympian   
1           1       Apollo    Ἀπόλλων  боженька  olympian   
2           2         Ares       Ἄρης  боженька  olympian   
3           3      Artemis    Ἄρτεμις  боженька  olympian   
4           4       Athena      Ἀθηνᾶ  боженька  olympian   

                                         description  
0      goddess of beauty, love, desire, and pleasure  
1  god of music, arts, knowledge, healing, plague...  
2       god of courage, war, bloodshed, and violence  
3  virgin goddess of the hunt, wilderness, animal...  
4  goddess of reason, wisdom, intelligence, skill...

Используя библиотеки Numpy и Pandas, а также файл CSV с датасетом при помощи read.csv и head мы выводим данный датасет.

In [2]:
greek_gods.shape 

(445, 6)

Размерность массива равна 445 строк на 5 столбцов, мэ это получили при помощи shape.

In [3]:
greek_gods.columns 

Index(['Unnamed: 0', 'name-english', 'name-greek', 'main-type', 'sub-type',
       'description'],
      dtype='object')

При помощи columns мы выводим названия всех колонок нашего датасета. У нас присутсвует айдишник, имя на английском, имя на греческом, типах, описание.

In [4]:
greek_gods.nunique()

Unnamed: 0      445
name-english    445
name-greek      441
main-type         3
sub-type         16
description     395
dtype: int64

Благодаря nunique мы получили все уникальные значения. 

In [5]:
greek_gods.sort_values(by = 'main-type')

Unnamed: 0 name-english name-greek        main-type         sub-type  \
145         145     Palioxis   Παλίωξις  personification  personification   
111         111     Euthenia    Εὐθενία  personification  personification   
112         112        Gelos      Γέλως  personification  personification   
113         113        Geras      Γῆρας  personification  personification   
114         114     Harmonia    Ἁρμονία  personification  personification   
..          ...          ...        ...              ...              ...   
237         237       Triton     Τρίτων         боженька              sea   
236         236      Triteia    Τριτεια         боженька              sea   
235         235       Thoosa      Θόοσα         боженька              sea   
242         242   Apheliotes  Αφηλιώτης         боженька              sky   
444         444        Damia      Δαμία         боженька            other   

                                           description  
145  spirit of backrush, flight and retreat from ba...  
111        spirit of prosperity, abundance, and plenty  
112                                 spirit of laughter  
113                                  spirit of old age  
114                     goddess of harmony and concord  
..                                                 ...  
237             fish-tailed son and herald of poseidon  
236           daughter of triton and companion of ares  
235                          goddess of swift currents  
242              god of the east wind (anemoi, άνεμοι)  
444                            local fertility goddess  

[445 rows x 6 columns]

При помощи sort.values мы вывели все значения в порядке возрастания type 

In [6]:
greek_gods['main-type'] = greek_gods['main-type'].replace('god' , 'боженька')

При помощи replace мы изменили все god в колонке main-type на Боженька

In [7]:
greek_gods.drop_duplicates()

Unnamed: 0 name-english name-greek main-type  sub-type  \
0             0    Aphrodite   Ἀφροδίτη  боженька  olympian   
1             1       Apollo    Ἀπόλλων  боженька  olympian   
2             2         Ares       Ἄρης  боженька  olympian   
3             3      Artemis    Ἄρτεμις  боженька  olympian   
4             4       Athena      Ἀθηνᾶ  боженька  olympian   
..          ...          ...        ...       ...       ...   
440         440       Rhapso       Ραψώ  боженька     nymph   
441         441    Alexiares  Αλεξιαρης  боженька     other   
442         442     Anicetus   Ανικητος  боженька     other   
443         443      Auxesia    Αυξησία  боженька     other   
444         444        Damia      Δαμία  боженька     other   

                                           description  
0        goddess of beauty, love, desire, and pleasure  
1    god of music, arts, knowledge, healing, plague...  
2         god of courage, war, bloodshed, and violence  
3    virgin goddess of the hunt, wilderness, animal...  
4    goddess of reason, wisdom, intelligence, skill...  
..                                                 ...  
440  minor goddess or nymph whose name apparently r...  
441  son of heracles presiding over the defence of ...  
442  son of heracles presiding over the defence of ...  
443                            local fertility goddess  
444                            local fertility goddess  

[445 rows x 6 columns]

При помощи drop_duplicates мы удалили дубликаты.

In [8]:
greek_gods.info() 


<class 'pandas.core.frame.DataFrame'>
RangeIndex: 445 entries, 0 to 444
Data columns (total 6 columns):
 #   Column        Non-Null Count  Dtype 
---  ------        --------------  ----- 
 0   Unnamed: 0    445 non-null    int64 
 1   name-english  445 non-null    object
 2   name-greek    445 non-null    object
 3   main-type     445 non-null    object
 4   sub-type      445 non-null    object
 5   description   445 non-null    object
dtypes: int64(1), object(5)
memory usage: 21.0+ KB


С помощью атрибута info мы получили общую информацию о датасете, включая количество непустых значений, типы данных каждого столбца, общее количество столбцов и количество использованной памяти. Мы видим, что всего 5 столбцов, количество использованной памяти = 17.5 + КB.

In [9]:
greek_gods.describe()

Unnamed: 0
count  445.000000
mean   222.000000
std    128.604691
min      0.000000
25%    111.000000
50%    222.000000
75%    333.000000
max    444.000000

При помощи describe мы вернули основные статистические показатели для каждого числового столбца, такие как количество, среднее значение, стандартное отклонение, минимальное и максимальное значение.

In [10]:
greek_gods.loc[:, 'main-type' : 'description']

main-type  sub-type                                        description
0    боженька  olympian      goddess of beauty, love, desire, and pleasure
1    боженька  olympian  god of music, arts, knowledge, healing, plague...
2    боженька  olympian       god of courage, war, bloodshed, and violence
3    боженька  olympian  virgin goddess of the hunt, wilderness, animal...
4    боженька  olympian  goddess of reason, wisdom, intelligence, skill...
..        ...       ...                                                ...
440  боженька     nymph  minor goddess or nymph whose name apparently r...
441  боженька     other  son of heracles presiding over the defence of ...
442  боженька     other  son of heracles presiding over the defence of ...
443  боженька     other                            local fertility goddess
444  боженька     other                            local fertility goddess

[445 rows x 3 columns]

При помощи loc мы вывели столбцы с main-type до description.

In [11]:
greek_gods.drop(columns = ['name-greek'])

Unnamed: 0 name-english main-type  sub-type  \
0             0    Aphrodite  боженька  olympian   
1             1       Apollo  боженька  olympian   
2             2         Ares  боженька  olympian   
3             3      Artemis  боженька  olympian   
4             4       Athena  боженька  olympian   
..          ...          ...       ...       ...   
440         440       Rhapso  боженька     nymph   
441         441    Alexiares  боженька     other   
442         442     Anicetus  боженька     other   
443         443      Auxesia  боженька     other   
444         444        Damia  боженька     other   

                                           description  
0        goddess of beauty, love, desire, and pleasure  
1    god of music, arts, knowledge, healing, plague...  
2         god of courage, war, bloodshed, and violence  
3    virgin goddess of the hunt, wilderness, animal...  
4    goddess of reason, wisdom, intelligence, skill...  
..                                                 ...  
440  minor goddess or nymph whose name apparently r...  
441  son of heracles presiding over the defence of ...  
442  son of heracles presiding over the defence of ...  
443                            local fertility goddess  
444                            local fertility goddess  

[445 rows x 5 columns]

При помощи drop мы удалили колонку name-greek. 

In [12]:
greek_gods.to_csv (r"C:\Users\user\Downloads\greek_gods2.csv")

При помощи to_csv мы сохранили наш датасет на рабочий стол.

Вывод: на данной практической работе я разбрала датасет о "Греческих богах". Он хранит в себе информацию о названии, дате выхода, стране и так далее сериала или фильма. С помощью определённых атрибутов узнала ифнормацию о данном датасете и проденстрировала это на деле.